### PR TITLE
Email admin updates

### PR DESCRIPTION
**Pull request by lynnco**
- add duplicate email name validator (this constraint was previous enforced by name being a primary key)
- add email sites display and filtering to admin

### CODE REVIEW

**Comment by lynnco** [comment]
I'll add a minor version release to this PR

File: tests/notification/test_validation.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from django.test import TestCase
+from django.core.exceptions import ValidationError
+
+from callisto.notification.models import EmailNotification
+
+
+class EmailValidationTest(TestCase):
+
+    def test_duplicate_emails_not_allowed(self):
```
Comment (kelseyq) [inline tests/notification/test_validation.py]: can we add a test that creates two duplicate-named emails on different sites, then tries to add the other site to one of them and verify that an error is also thrown then?

Comment (lynnco) [inline tests/notification/test_validation.py]: 👍

Comment (lynnco) [inline tests/notification/test_validation.py]: glad you poked me to add more tests, because I the test I wrote wasn't testing the logic well enough
